core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -310,11 +311,37 @@ public URIStatus getStatus(AlluxioURI path)
   public URIStatus getStatus(AlluxioURI path, final GetStatusPOptions options)
       throws FileDoesNotExistException, IOException, AlluxioException {
     checkUri(path);
-    return rpc(client -> {
-      GetStatusPOptions mergedOptions = FileSystemOptions.getStatusDefaults(
-          mFsContext.getPathConf(path)).toBuilder().mergeFrom(options).build();
-      return client.getStatus(path, mergedOptions);
-    });
+    return getStatusThroughRPC(path, options);
+  }
+
+  /**
+   * Gets status of the path when openFile.
+   *
+   * If {@link OpenFilePOptions#getUpdateLastAccessTime()} is true,
+   * then update the path's last access time in the getStatus RPC.
+   *
+   * @param path the path
+   * @param options the open file options
+   * @return the status for path
+   * @throws FileDoesNotExistException when path does not exist
+   * @throws OpenDirectoryException when path is a directory
+   * @throws FileIncompleteException when path is a file and is not completed yet
+   */
+  protected URIStatus getStatus(AlluxioURI path, OpenFilePOptions options)
```

Comment:
Do we need this variant of `getStatus` implemented in `BaseFileSystem` (or ever?), could the caller create the getStatus options and check for the two new exceptions?

Reply:
it's just a utility function exposed for reducing duplicated codes